next.config.js

```diff
@@ -0,0 +1,164 @@
+            },
+        }],
+    ], {
+        webpack: (config, { dev, isServer }) => {
+            const { defaultConfig: { assetPrefix } } = nextConfig;
+
+            // Setup env variables that will be available in the client-side
+            {
+                const envVars = pickBy(process.env, (value, key) => key.startsWith('REACT_APP_'));
+
+                config.plugins.push(new webpack.EnvironmentPlugin(envVars));
+            }
+
+            // Remove exclude condition to transpile every node_module
+            {
+                const transpilingRule = config.module.rules.find((rule) => rule.use.loader === 'next-babel-loader');
+
+                delete transpilingRule.exclude;
+            }
+
+            // Raster images (png, jpg, etc)
+            config.module.rules.push({
+                test: /\.(png|jpg|jpeg|gif|webp)$/,
+                exclude: /favicons\/.*$/,
+                loader: require.resolve('file-loader'),
+                options: {
+                    name: dev ? 'images/[name].[ext]' : 'images/[name].[hash:15].[ext]',
+                    publicPath: `${assetPrefix}/_next/static/chunks/media`,
+                    outputPath: 'static/chunks/media',
+                    emitFile: !isServer,
+                },
+            });
+
+            // Favicons used in the server
+            config.module.rules.push({
+                test: /favicons\/.*$/,
+                loader: require.resolve('file-loader'),
+                options: {
+                    name: dev ? 'favicons/[name].[ext]' : 'favicons/[name].[hash:15].[ext]',
+                    publicPath: `${assetPrefix}/_next/static/chunks/media`,
+                    outputPath: '../static/chunks/media',
+                    emitFile: isServer,
+                },
+            });
+
+            // Web fonts
+            config.module.rules.push({
+                test: /\.(eot|ttf|woff|woff2|otf)$/,
+                loader: require.resolve('file-loader'),
+                options: {
+                    name: dev ? 'fonts/[name].[ext]' : 'fonts/[name].[hash:15].[ext]',
+                    publicPath: `${assetPrefix}/_next/static/chunks/media`,
+                    outputPath: 'static/chunks/media',
+                    emitFile: !isServer,
+                },
+            });
+
+            // Audio & Video
+            config.module.rules.push({
+                test: /\.(mp3|flac|wav|aac|ogg|oga|mp4|m4a|webm|ogv)$/,
+                loader: require.resolve('file-loader'),
+                options: {
+                    name: dev ? 'playback/[name].[ext]' : 'playback/[name].[hash:15].[ext]',
+                    publicPath: `${assetPrefix}/_next/static/chunks/media`,
+                    outputPath: 'static/chunks/media',
+                    emitFile: !isServer,
+                },
+            });
+
+            // SVG files
+            config.module.rules.push({
+                test: /images\/.*.svg$/,
```

Comment:
In here we're forcing the users to use the folder `images/` to put SVGs that aren't considered to be inline. In conversation with @satazor we referenced that we shouldn't use folder specific rules but file specific rules (since we're dealing basically with file extensions).
With that said, and with the developments made in [@moxy/next-common-files](https://github.com/moxystudio/next-common-files), I think it's better to have something like `.data-url.svg` for "normal" SVGs and use `.svg` for inline SVGs. Or another name.